sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -5292,6 +5292,183 @@ object functions {
    */
   def hours(e: Column): Column = withExpr { Hours(e.expr) }
 
+  /**
+   * Creates a map after splitting the text into key/value pairs using delimiters.
+   * Both `pairDelim` and `keyValueDelim` are treated as regular expressions.
+   *
+   * @group map_funcs
+   * @since 3.5.0
+   */
+  def str_to_map(text: Column, pairDelim: Column, keyValueDelim: Column): Column = withExpr {
```

Comment:
hmm... does these new functions have test coverage?

Reply:
Oops！I forget to add it, let me add right now.

Reply:
or you can add more doctest, which will be also used in python client.

Reply:
OK, I will add both.